step-definitions/Hotels/hotelsSD2.js

```diff
@@ -0,0 +1,177 @@
+const { ANDROID_CONFIG } = require("@wdio/cli/build/constants");
+const { Given, When, Then } = require("@wdio/cucumber-framework");
+const { expect } = require("chai");
+const Homepage2 = require("../../Pages/Hotels/Homepage2");
+const Listyourproperty = require("../../Pages/Hotels/Listyourproperty");
+const Searchpage = require("../../Pages/Hotels/Searchpage");
+
+const hpage2 = new Homepage2();
+const srchpage = new Searchpage();
+const lyppage = new Listyourproperty();
+// TC-30
+When(/^I scroll to "Get the app" button$/, async function(){
+    await hpage2.scrollToGetTheApp();
+})
+When(/^I enter “0000000000” in phone number$/, async function(){
+    await hpage2.enterPhoneNumber('0000000000');
+})
+When(/^I click on “Get the app“ button$/, async function(){
+    await hpage2.clickGetTheApp();
+})
+Then(/^I verify “Please enter a valid phone number.“ error is displayed$/, async function(){
+    expect(await hpage2.phoneNumErrMsgIsDisplayed(), 'Error message not displayed').to.be.true;
+})
+//TC-17
+When(/^I click on dates$/, async function(){
+    await hpage2.clickDatesBtn();
+})
+When(/^I go to current month if date not displayed$/, async function(){
+    await hpage2.goToCurrentMonth();
+}) 
+Then(/^I verify for current month past dates, if any, are disabled$/, async function(){
+    expect(await hpage2.verifyPastDatesDisabled(), 'Dates not disabled').to.be.true;
+})
+Then(/^I verify back button on current month is disabled$/, async function(){
+    expect(await hpage2.verifyCuurentMonthBackBtnDisabled(), 'Button not disabled').to.be.false;
+})
+//TC-19
+When(/^I enter "(bora|Manhattan, NY)" in destination$/, async function(fieldname){
+    switch(fieldname) {
+        case("bora"):
+            await hpage2.enterDestination('bora');
+            break;
+        case("Manhattan, NY"):
+            await hpage2.enterDestination('Manhattan, NY');
+            await hpage2.selectDestinationFromAutoSuggestion('manhattan');
+            break;
+    }
+})
+When(/^I select “Bora Bora, Leeward Islands, French Polynesia” from auto suggestion$/, async function(){
+    await hpage2.selectDestinationFromAutoSuggestion('bora bora');
+})
+When(/^I select Dec 1, 2022 as Check-in$/, async function(){
+    await hpage2.selectCheckInDate('December', '2022', '1');
```

Comment:
Its better and dynamic to pass all data like date/mob num/destination etc as parameter from feature file.